src/cmd/build.rs

```diff
@@ -195,56 +217,130 @@ impl BuildCommand {
             verbosity = Verbosity::Quiet;
         }
 
-        let args = ExecuteArgs {
-            manifest_path,
-            verbosity,
-            build_mode,
-            network,
-            build_artifact: self.build_artifact,
-            unstable_flags,
-            optimization_passes,
-            keep_debug_symbols: self.keep_debug_symbols,
-            skip_linting: self.skip_linting,
-            output_type,
-        };
+        let mut build_results = Vec::new();
 
-        execute(args)
+        match self.build_all {
+            true => {
+                let workspace_members = get_cargo_workspace_members(&manifest_path)?;
+                let mut args = ExecuteArgs {
+                    verbosity,
+                    build_mode,
+                    network,
+                    build_artifact: self.build_artifact,
+                    unstable_flags: unstable_flags.clone(),
+                    optimization_passes,
+                    keep_debug_symbols: self.keep_debug_symbols,
+                    skip_linting: self.skip_linting,
+                    output_type: output_type.clone(),
+                    ..Default::default()
+                };
+                for package_id in workspace_members {
+                    args.manifest_path =
+                        util::extract_subcontract_manifest_path(package_id)
+                            .expect("Error extracting package manifest path");
+                    build_results.push(execute(args.clone())?);
+                }
+            }
+            false => {
+                let args = ExecuteArgs {
+                    manifest_path,
+                    verbosity,
+                    build_mode,
+                    network,
+                    build_artifact: self.build_artifact,
+                    unstable_flags,
+                    optimization_passes,
+                    keep_debug_symbols: self.keep_debug_symbols,
+                    skip_linting: self.skip_linting,
+                    output_type,
+                };
+                build_results.push(execute(args)?);
+            }
+        }
+
+        Ok(build_results)
```

Comment:
This would be a for a follow up PR, but it could be cool to have these results aggregated in a table.

E.g 


| Contract         | Size         | Metadata Path |
|--------------|-----------|----------------|
| Foo                | 24kB | target/ink/foo           | 
| Bar                | 64kB |  target/ink/bar          |

Reply:
should a new issue be created for that?

Reply:
Yeah go for it

Reply:
Filed in https://github.com/paritytech/cargo-contract/issues/727